src/stream.jl

```diff
@@ -489,21 +489,21 @@ function Base.write(stream::TranscodingStream)
     return 0
 end
 
-function Base.write(stream::TranscodingStream, b::UInt8)
+function Base.write(stream::TranscodingStream, b::UInt8)::Int
     changemode!(stream, :write)
-    if marginsize(stream.buffer1) == 0 && flushbuffer(stream) == 0
-        return 0
-    end
-    return writebyte!(stream.buffer1, b)
+    buffer1 = stream.buffer1
+    marginsize(buffer1) > 0 || flush_buffer1(stream)
+    writebyte!(buffer1, b)
+    return 1
 end
 
 function Base.unsafe_write(stream::TranscodingStream, input::Ptr{UInt8}, nbytes::UInt)
     changemode!(stream, :write)
-    state = stream.state
     buffer1 = stream.buffer1
     p = input
     p_end = p + nbytes
-    while p < p_end && (marginsize(buffer1) > 0 || flushbuffer(stream) > 0)
+    while p < p_end
+        marginsize(buffer1) > 0 || flush_buffer1(stream)
```

Comment:
`flush_buffer1` must empty `buffer1` or error, so this loop should make progress.